obspy/io/gcf/__init__.py

```diff
@@ -18,11 +19,83 @@
 The format will be determined automatically.
 
 >>> from obspy import read
->>> st = read("/path/to/20160603_1955n.gcf")
+>>> st = read("/path/to/20160603_1955n.gcf")  #doctest: +ELLIPSIS
 
-You can also specify the following keyword arguments that change the
-behavior of reading the file:
-* ``headonly=True``: Read only the header part, not the actual data
+GCF specific metadata is stored in ``stats.gcf``
+
+>>> for k, v in sorted(st[0].stats.gcf.items()):
+...     print("'%s': %s" % (k, str(v)))  #doctest: +NORMALIZE_WHITESPACE
+'FIC': -49378
+'RIC': -49312
+'blk': 0
+'digi': 0
+'gain': 1
+'stat': 0
+'streamID': 6018N4
+'sysType': 1
+'systemID': 6281
+'t_leap': False
+'ttl': 6
+
+Several keyword arguments are available to set parameters required by the returned 
+:class:`~obspy.core.stream.Stream` object as well as to control merging of data blocks
+on reading: ``networkcode``, ``stationcode``, ``locationcode``, ``bandcode``, 
+``instrumentcode``, ``channel_prefix``, ``blockmerge``, ``headonly``, 
+``cleanoverlap``, ``errorret``. They are passed to the :meth:`~obspy.io.gcf.core._read_gcf` 
+method so refer to this for details of each parameter.
+
+Function may raise TypeError on bad arguments (e.g. out of range) and IOError on failure
+to read/decode file or errors in data blocks in file (bad blocks may be returned by use
+of keyword argument ``errorret`` though, preventing IOError's raised by these)
+
+Writing
+-------
+Write data back to disc or a file like object using the
+:meth:`~obspy.core.stream.Stream.write` method of a
+:class:`~obspy.core.stream.Stream` or
+:class:`~obspy.core.trace.Trace` object.
+
+>>> st.write('GCF-filename.gcf', format='GCF') #doctest: +SKIP
+
+
+Several key word arguments are available to set required GCF specific header information and
+allowed missalignment of supported starttime of first data sample:
+``streamID``, ``systemID``, ``isLeap``, ``gain``, ``ttl``, ``digi``, ``sysType``, ``missalign``
+
+GCF specific header information can also be provided in ``stats.gcf`` on each
+:class:`~obspy.core.trace.Trace` object. If a specific piece of header information is available
+both in ``stats.gcf`` and set by its correspondng keyworde argument, the keyword argument will 
+have precedenc. For default values if neither set by ``stats.gcf`` nor by use of appropriate 
+keyword arguments refer to :meth:`~obspy.io.gcf.core._read_gcf`
+
+..Note:: 
+The GCF format is only guaranteed to support 32-bit signed integer values. While data with values
+out of range may be properly stored in the GCF format (if first and last data sample can be represented
+as a 32-bit signed integer as well as all first difference values of the dat avector) the current 
+implementaion only permitts input data to be representable as a 32-bit signed integer. If input waveforms
+cannot be representable as 32-bit signed integers they will silently be clipped at -2,147,483,648 and 
+2,147,483,647
```

Comment:
It would actually be trivial to add a warning message at that part where you crop to max/min.

```
data = np.ceil(tr.data, ...)
data = np.floor(data, ...)
data = np.require(data, dtype=np.int32, requirements=['C_CONTIGUOUS'])
if not np.all(data == tr.data):
    msg = '...'
    warnings.warn(msg)
tr.data = data
```

Reply:
Sure, will do

Reply:
fixed. also now not showing a clipping warning but a correct warning in case of no clipping but losing decimal info when downcasting floats